Blish HUD/GameServices/Overlay/DynamicHUDMethod.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.ComponentModel;
+
+namespace Blish_HUD.Graphics {
+    public enum DynamicHUDMethod : int {
+        [Description("Always Show (Default)")]
+        AlwaysShow = 0,
+
+        [Description("Show Only out of Combat")]
+        ShowPeaceful = 1,
+
+        [Description("Show Only in Combat")]
+        ShowInCombat = 2,
```

Comment:
Could we perhaps do `SetExcluded` for the current setting and this enum value?  I like the idea of having this as an option for other items, but having just gotten myself stuck with it set like this and having to go back into combat to change the setting back was kind of a weird experience.

I'm hoping it's safe to assume that there aren't too many people out there that would want this for the corner icons (but again, good to keep as we add support for other UI elements).